src/components/PopupWithForm.js

```diff
@@ -17,11 +37,18 @@ export default class PopupWithForm extends Popup {
     return inputvalues;
   }
 
+  setInputs(inputValues) {
+    this.inputs.forEach((input) => {
+      input.value = inputValues[input.name];
+    });
+  }
+
   setEventListeners() {
     super.setEventListeners();
     this.form.addEventListener("submit", (e) => {
       e.preventDefault();
-      this.submit(e, this._getInputValues());
+      this.renderLoading(true);
+      new Promise((resolve) => resolve(this.submit(e, this._getInputValues())));
```

Comment:
Needs correction: 

1) there is no need to create a promise here, just call `this.submit`
2) this.submit should have only one argument - the input values